launcher/ui/dialogs/ExportInstanceDialog.cpp

```diff
@@ -46,301 +46,21 @@
-                    {
-                        // or just block this one.
-                        blocked.insert(relpath);
-                    }
-                    row++;
-                }
-            }
-            changed = true;
-        }
-        if (changed)
-        {
-            // update the thing
-            emit dataChanged(index, index, {Qt::CheckStateRole});
-            // update everything above index
-            QModelIndex up = index.parent();
-            while (1)
-            {
-                if (!up.isValid())
-                    break;
-                emit dataChanged(up, up, {Qt::CheckStateRole});
-                up = up.parent();
-            }
-            // and everything below the index
-            QModelIndex doing = index;
-            int row = 0;
-            QStack<QModelIndex> todo;
-            while (1)
-            {
-                auto node = this->index(row, 0, doing);
-                if (!node.isValid())
-                {
-                    if (!todo.size())
-                    {
-                        break;
-                    }
-                    else
-                    {
-                        doing = todo.pop();
-                        row = 0;
-                        continue;
-                    }
-                }
-                emit dataChanged(node, node, {Qt::CheckStateRole});
-                todo.push(node);
-                row++;
-            }
-            // siblings and unrelated nodes are ignored
-        }
-        return true;
-    }
-
-    bool shouldExpand(QModelIndex index)
-    {
-        QModelIndex sourceIndex = mapToSource(index);
-        QFileSystemModel *fsm = qobject_cast<QFileSystemModel *>(sourceModel());
-        if (!fsm)
-        {
-            return false;
-        }
-        auto blockedPath = relPath(fsm->filePath(sourceIndex));
-        auto found = blocked.find(blockedPath);
-        if(found)
-        {
-            return !found->leaf();
-        }
-        return false;
-    }
-
-    void setBlockedPaths(QStringList paths)
-    {
-        beginResetModel();
-        blocked.clear();
-        blocked.insert(paths);
-        endResetModel();
-    }
-
-    const SeparatorPrefixTree<'/'> & blockedPaths() const
-    {
-        return blocked;
-    }
-
-protected:
-    bool filterAcceptsColumn(int source_column, const QModelIndex &source_parent) const
-    {
-        Q_UNUSED(source_parent)
-
-        // adjust the columns you want to filter out here
-        // return false for those that will be hidden
-        if (source_column == 2 || source_column == 3)
-            return false;
-
-        return true;
-    }
-
-private:
-    InstancePtr m_instance;
-    SeparatorPrefixTree<'/'> blocked;
-};
-
 ExportInstanceDialog::ExportInstanceDialog(InstancePtr instance, QWidget *parent)
     : QDialog(parent), ui(new Ui::ExportInstanceDialog), m_instance(instance)
 {
     ui->setupUi(this);
     auto model = new QFileSystemModel(this);
```

Comment:
The solution to the Icon provider slow down is over here, not in the proxy model.

 this should preserve dir icons
 At the top of the file add
```cpp
#include <QFileIconProvider>
class EmptyIconProvider : public QFileIconProvider {
   public:
    // in theroy the IconType overload is safe, it gets these strait form the theme
    // QIcon icon(QAbstractFileIconProvider::IconType type) const override { return {}; };
    QIcon icon(const QFileInfo& info) const override { 
        if (info.isDir()) // preserve dir icons;
            return QFileIconProvider::icon(info);
        return {}; 
    };
};
```
then add this line

```suggestion
    auto model = new QFileSystemModel(this);
    model->setIconProvider(new EmptyIconProvider());
```

NOTE: I DO NOT know if this will leak the EmptyIconProvider
it may be best to give the page an instance as a member and pass a pointer to *that* is instead. (or, if for some reason having it as a raw member causes issues make the member a `shared_qobject_ptr` set up in the constructor)

@flowln  would you mind testing if this fixes the lag issues? I know the empty provider works but I don't get the lag issue locally.

Reply:
I have a feeling something changed in Qt 6's implementation. Maybe there could be an implementation which uses Prism's icons instead of the system icons since that'd look better anyway?

Reply:
I don't think we have file/folder icons on prism icon themes rn

Reply:
Yes, I built with Qt 5 and it had much better performance! Reimplementing the icon provider should have the same effect? Probably?

I'm pretty sure that'd be a leak since you're not setting the QObject's parent? Assuming it's a QObject :p

Reply:
> I don't think we have file/folder icons on prism icon themes rn

Oops, i kind of missed that. `viewfolder and `log` look good.

Reply:
Okay, I made an optimised icon provider... but it has no improvement on kde where it's not laggy at all